src/domain/safe/safe.repository.interface.ts

```diff
@@ -48,16 +48,37 @@ export interface ISafeRepository {
     offset?: number,
   ): Promise<Page<ModuleTransaction>>;
 
-  getTransactionQueueByModified(
+  /**
+   * Returns the Safe Transaction Queue in ascending order.
+   *
+   * The ascending order is first considered for the nonce.
+   * If multiple transactions have the same nonce, then they
+   * will be sorted according to the submission date
+   *
+   * @param chainId - the chain id from where to retrieve the transactions from
+   * @param safe - the target safe from where to get the transaction from
+   * @param limit - the maximum number of transactions to be returned
+   * @param offset - the starting point from which to start the pagination
+   */
+  getTransactionQueue(
     chainId: string,
-    safeAddress: string,
+    safe: Safe,
     limit?: number,
     offset?: number,
   ): Promise<Page<MultisigTransaction>>;
 
-  getTransactionQueueByNonce(
+  /**
+   * Returns the Safe Transaction Queue sorted by modified status.
+   * (from most recently modified to last)
+   *
+   * @param chainId - the chain id from where to retrieve the transactions from
+   * @param safe - the target safe from where to get the transaction from
+   * @param limit - the maximum number of transactions to be returned
+   * @param offset - the starting point from which to start the pagination
+   */
+  getTransactionQueueByModified(
```

Comment:
I agree this simplifies the repository API, but I'm also a bit concerned about the fact that changing the primitive input parameters (`safeAddress: string, nonce: number`) by a domain object (`safe: Safe`) forces the client of the repository to retrieve the `Safe` beforehand. I think it's acceptable in this context, which is limited to a concrete feature, just commenting as a side note 🙂

Reply:
For the use cases that I have seen we always have the `SafeInfo`. In my opinion by using the domain object itself makes it more robust (not only in the function signature by not using `string` but also to use other safe properties that are available if needed).

Adding the `nonce` on top of the `safeAddress` seemed a bit odd in this context because we actually want the `nonce` of the `safeAddress`.

Reply:
Yes, after giving it a second thought, it makes sense 👍